platform/app/src/routes/DataSourceWrapper.tsx

```diff
@@ -66,7 +95,51 @@ function DataSourceWrapper(props) {
   const [data, setData] = useState(DEFAULT_DATA);
   const [isLoading, setIsLoading] = useState(false);
 
+  /**
+   * The effect to initialize the data source whenever it changes. Similar to
+   * whenever a different Mode is entered, the Mode's data source is initiliazed, so
+   * too this DataSourceWrapper must initialize its data source whenever a different
+   * data source is activated. Furthermore, a data source might be initialized
+   * several times as it gets activated/deactivated because the location URL
+   * might change and data sources initialize based on the URL.
+   */
+  useEffect(() => {
+    const initializeDataSource = async () => {
+      await dataSource.initialize({ params, query });
+      setIsDataSourceInitialized(true);
+    };
+
+    initializeDataSource();
+  }, [dataSource]);
+
   useEffect(() => {
+    const dataSourceChangedCallback = () => {
```

Comment:
when a datasource is changed I thought we should do `const studies = await dataSource.query.studies.search(queryFilterValues);` again, no?

Reply:
The call to `setData(DEFAULT_DATA)` triggers the refresh that you mention. So yes it does that indirectly.

Reply:
I think it will not trigger that useEffect if the same data source has changed its configuration, or will it? which dependency will make it re run?

Reply:
Yes it will. I have tested it. For just a config change, `setData(DEFAULT_DATA)` does it.